### PR TITLE
NEXUS-8892 Maven2 query improvements

### PR DESCRIPTION
**Pull request by mrprescott**
Maven2 had a number of 'key' fields that concatenated relevant info into a single field. Since the values are wholly derived from CMA-standard indexed fields (group, name, version), it's (measurably) faster to just perform queries against those and drop the key columns.

(There's a possibility this overlooks some corner case of the key field usage, be great if @cstamas could weigh in on Monday. Marking POC until then.)

Issue
https://issues.sonatype.org/browse/NEXUS-8892

CI
http://bamboo.s/browse/NX3-OSSF560

### CODE REVIEW

**Comment by kellyrob99** [comment]
seems sensible +1

File: plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java [hunk 1]
```diff
@@ -365,9 +356,20 @@ private Component findComponent(final StorageTx tx,
                                   final Bucket bucket,
```
Comment (sonatype-zion) [inline plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java]: MAJOR Avoid unused method parameters such as 'bucket'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=pmd%3AUnusedFormalParameter)

**Comment by sonatype-zion** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Private method 'getComponentKey' is never used.](https://github.com/sonatype/nexus-oss/blob/67b3af596dede7be044968e6427bcd57c0756557/plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java#L345) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
* MAJOR [Class cannot be instantiated and does not provide any static methods or fields](https://github.com/sonatype/nexus-oss/blob/67b3af596dede7be044968e6427bcd57c0756557/components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/ComponentQuery.java#L28) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=pmd%3AMissingStaticMethodInNonInstantiatableClass)

**Comment by mrprescott** [comment]
'Class cannot be instantiated' warning is false positive.

**Comment by cstamas** [comment]
+1 and this change lesses redundant payload of records too.

**Comment by mcculls** [comment]
+1

**Comment by sonatype-zion** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Class cannot be instantiated and does not provide any static methods or fields](https://github.com/sonatype/nexus-oss/blob/c21785a2ce263c6f732b6ebd05d757e95da193ae/components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/ComponentQuery.java#L28) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=pmd%3AMissingStaticMethodInNonInstantiatableClass)

**Comment by cstamas** [comment]
http://bamboo.s/browse/NX3-OSSF560-3 :+1:

**Comment by mrprescott** [comment]
Merged 410a7e3

**Pull request closed without merging**